odoo/custom/src/private/nxpo_procurement_type/wizards/purchase_request_line_make_purchase_order.py

```diff
@@ -0,0 +1,48 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class PurchaseRequestLineMakePurchaseOrder(models.TransientModel):
+    _inherit = "purchase.request.line.make.purchase.order"
+
+    procurement_type = fields.Many2one(
+        comodel_name="procurement.type",
+        string="Procurement Type",
+    )
+
+    @api.model
+    def default_get(self, fields):
+        res = super().default_get(fields)
+        active_model = self.env.context.get("active_model", False)
+        if active_model == "purchase.request.line":
+            procurement_type = (
+                self.env[active_model]
+                .browse(self.env.context.get("active_ids", False))
+                .mapped("request_id.procurement_type")
+                .ids
+            )
+        elif active_model == "purchase.request":
+            procurement_type = (
+                self.env[active_model]
+                .browse(self.env.context.get("active_ids", False))
+                .mapped("procurement_type")
+                .ids
+            )
+
+        if len(procurement_type) == 1:
+            res["procurement_type"] = (
+                self.env["procurement.type"].browse(procurement_type).id
+            )
```

Comment:
This is redundant, why you .browse(id) and then get the id again?
```suggestion
            res["procurement_type"] = procurement_type_id
```